test/groupinterface.jl

```diff
@@ -0,0 +1,58 @@
+using Interfaces
+using Test
+
+module Group
+
+using Interfaces
+
+function neutral end
+function multiplication end
+function inversion end
+
+@interface GroupInterface (
+    mandatory = (;
+        neutral_check = (
+            "neutral stable" => a::Arguments -> neutral(a.x) isa typeof(a.x),
+        ),
+        multiplication_check = (
+            "multiplication stable" => a::Arguments -> multiplication(a.x, a.y) isa typeof(a.x),
+        ),
+        inversion_check = (
+            "inversion stable" => a::Arguments -> inversion(a.x) isa typeof(a.x),
+            "inversion works" => a::Arguments -> multiplication(a.x, inversion(a.x)) ≈ neutral(typeof(a.x)),
+        ),
+    ),
+    optional = (;
+        abelian = (
+            "multiplication commutes" => a::Arguments -> multiplication(a.x, a.y) ≈ multiplication(a.y, a.x),
+        ),
+    )
+) """
+A group is a set of elements with a neutral element where you can perform multiplications and inversions.
```

Comment:
It would be nice to demonstrate documenting the required `Arguments` here

Reply:
Yeah, I just would like to wait until the docs website is back and running, to set up a Literate pipeline that removes the need for copy paste. What do you think?

Reply:
Basically that way we could put the docs in the test file

Reply:
Done. Looks like I just added the DOCUMENTER_KEY a year ago and forgot about the deploy keys :sweat_smile:

Reply:
I havent used Literate.jl like that so interested to see what you do.

Reply:
I'll do it in a separate PR for the existing code first, to avoid bloating this one

Reply:
#17